### PR TITLE
Fix Chrome issues with beep and alert

### PR DESCRIPTION
**Pull request by sketchytechky**
In Chrome, e.ggtimer now do not beep until the alert box has been acknowledged.  This is troublesome since the alert box happen in background and there's no way to alert the user that time was up.

## Why?

This was not the case previously.  In Chrome 50 introduce a change which return a [promise](https://developers.google.com/web/updates/2016/03/play-returns-promise) on `audio.play()`. 
 Unfortunately this introduce and interesting side effect that the beep will not happen until the alert box have been acknowledged.

## Fix

This PR introduce the use of the promise (when exists - e.g. in Safari, `audio.play()` returns `undefined` - which this code also handles.

Unfortunately I do not know what minifies was being used in the original build.  I have used [UglifyJS](https://github.com/mishoo/UglifyJS) to minify the code.

### CODE REVIEW

File: pages/scripts/Egg3.js [hunk 1]
```diff
@@ -137,19 +137,27 @@ var Egg = {
...
         }
 
         if (Egg.sequence.length === 0) {
             clearInterval(Egg.ticker);
             Egg.updateTitle(Egg.expiredMessage);
             Egg.updateText(Egg.expiredMessage);
-            Egg.showAlert();
+
+            // Let the beep finish before showing alert
+            if (beepFinishedPromise) {
```
Comment (lemieuxster) [inline pages/scripts/Egg3.js]: Out of an abundance of caution, can this also check that the type is `Promise` or that is at least has a `then` method? 

```
if (beepFinishedPromise && (beepFinishedPromise instanceof Promise || 'then
 in beepFinishedPromise)) 
```

**Comment by lemieuxster** [comment]
I use Uglify as well. I should include stuff in package.json and build scripts.

File: pages/scripts/Egg3.js (hunk 1 above)
Comment (sketchytechky) [reply pages/scripts/Egg3.js]: Thanks! Good call - did a bit of look around https://stackoverflow.com/questions/27746304/how-do-i-tell-if-an-object-is-a-promise

`obj && typeof obj.then == 'function'` seems like the winner here.

**Comment by lemieuxster** [comment]
Pushed this live. Thanks again!

**Comment by sketchytechky** [comment]
Thanks for the speedy pushed 👍  - tested live and confirmed that it is fixed.

**Pull request merged**